### DISCUSSION TITLE
Return values for prisma.$transaction

### DISCUSSION THREAD

**Discussion by mkoo21** [prisma/prisma#3800] [Q&A] [answered]
If I am creating a user, I can get the ORM representation of the newly created user as the value `prisma.user.create` resolves to. 

Suppose I am creating a user as part of a transaction like
```ts
const write1 = prisma.user.create()
const write2 = prisma.post.create()
const write3 = prisma.profile.create()
await prisma.$transaction([write1, write2, write3])
```
How can I get e.g. the `id` of the user created in `write1`?

**Comment by ryands17** [comment, marked as answer]
@mkoo21 👋 

Do you want to use the `id` in `write1` obtained for other transactions? If so that is not possible right niow. It's better to use [nested writes](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/relation-queries#nested-writes) if you want to add relational data.

If you just need the output, you can get it directly like this:

```ts
const [user,,] = await prisma.$transaction([write1, write2, write3])
```

**Reply by mkoo21** [reply]
Immediately I just wanted to return the created object in the API response so that it could be used later. But it seems like we just have to hold out for better transaction support.

**Reply by ryands17** [reply]
Did you want to return it like this in the API?

```ts
return await prisma.$transaction([write1, write2, write3])[0].id
```

**Comment by mrsupon** [comment]
await prisma.$transaction( async( p )=>{  
  const newUser = await p.user.create( ... );
  // can use newUser attributes 
  const newPost = await p.post.create( ... );
  const newProfile = await p.profile.create( ... );
} );

ref: https://www.prisma.io/docs/orm/prisma-client/queries/transactions

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
